pygments/lexers/wren.py

```diff
@@ -0,0 +1,97 @@
+"""
+    pygments.lexers.wren
+    ~~~~~~~~~~~~~~~~~~~~
+    Lexer for Wren.
+    :copyright: Copyright 2006-2022 by the Pygments team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+import re
+
+from pygments.lexer import RegexLexer
+from pygments.token import Whitespace, Punctuation, Keyword, Name, Comment, \
+    Operator, Number, String
+
+__all__ = ['WrenLexer']
+
+class WrenLexer(RegexLexer):
+    name = 'Wren'
+    aliases = ['wren']
+    filenames = ['*.wren']
+
+    flags = re.MULTILINE | re.DOTALL
+
+    tokens = {
+        'root': [
+            # Whitespace.
+            (r'\s+', Whitespace),
+            (r'[,\\\[\]{}]', Punctuation),
+
+            # Push a parenthesized state so that we know the corresponding ')'
+            # is for a parenthesized expression and not interpolation.
+            (r'\(', Punctuation, ('parenthesized', 'root')),
+
+            # In this state, we don't know whether a closing ')' is for a
+            # parenthesized expression or the end of an interpolation. So, do
+            # a non-consuming match and let the parent state (either
+            # 'parenthesized' or 'interpolation') decide.
+            (r'(?=\))', Whitespace, '#pop'),
+
+            # Keywords.
+            (r'(as|break|class|construct|continue|else|for|foreign|if|import|'
+             r'in|is|return|static|super|var|while)\b', Keyword),
+
+            (r'(true|false|null)\b', Keyword.Constant),
+
+            (r'this\b', Name.Builtin),
+
+            # Comments.
+            (r'/\*', Comment.Multiline, 'comment'), # Multiline, can nest.
+            (r'//.*?$', Comment.Single),            # Single line.
+            (r'#.*?(\(.*?\))?$', Comment.Special),  # Attribute or shebang.
+
+            # Names and operators.
+            (r'[!%&*+\-./:<=>?\\^|~]+', Operator),
+            (r'[a-z][a-zA-Z_0-9]*', Name),
+            (r'[A-Z][a-zA-Z_0-9]*', Name.Class),
+            (r'__[a-zA-Z_0-9]*', Name.Variable.Class),
+            (r'_[a-zA-Z_0-9]*', Name.Variable.Instance),
+
+            # Numbers.
+            (r'0x[0-9a-fA-F]+', Number.Hex),
+            (r'\d+(\.\d+)?([eE][-+]?\d+)?', Number.Float),
+
+            # Strings.
+            (r'"""', String, 'raw_string'),
+            (r'"', String, 'string'),
+        ],
+        'comment': [
+            (r'/\*', Comment.Multiline, '#push'),
+            (r'\*/', Comment.Multiline, '#pop'),
+            (r'.', Comment.Multiline), # All other characters.
```

Comment:
This should be:
```python
'comment': [
    (r'\*/', Comment.Multiline, '#pop'),
    (r'[^*]+', Comment.Multiline),
    (r'\*', Comment.Multiline),
]
```
See: https://github.com/pygments/pygments/blob/master/Contributing.md

Reply:
Changed, thanks.

Reply:
Unlike many other C-family languages, block  comments in Wren can nest (see [here](https://wren.io/syntax.html#Comments)).

Although it's not a big problem, your standard pattern for block comments doesn't deal properly with that. So, if I try to highight this line:
```javascript
/* This is /* a nested */ comment. */
```
the word 'comment' is highlighted as a name rather than as part of the overall comment because the lexer thinks that the first ```*/``` is the end of the outer block.

Do you have a suggested pattern for that situation? If not, I'm happy to leave it as it is.

Reply:
OK, thanks I've changed it to that. Nested block comments now working fine :)